kimchi/src/circuits/expr.rs

```diff
@@ -1257,6 +1295,17 @@ impl<F: FftField> Expr<ConstantExpr<F>> {
                     }
                 }
             }
+            Expr::EnabledIf(feature, e) => {
+                let tok = PolishToken::SkipIf(*feature, 0);
+                res.push(tok);
+                let len_before = res.len();
+                /* Clone the cache, to make sure we don't try to access cached statements later
+                when the feature flag is off. */
+                let mut cache = cache.clone();
+                e.to_polish_(&mut cache, res);
+                let len_after = res.len();
+                res[len_before - 1] = PolishToken::SkipIf(*feature, len_after - len_before);
```

Comment:
I would find it cleaner to create a temporary vec, and then extend `res` once we know the length. I always find code that goes back to modify a buffer a bit scary.